components/dashboard/src/admin/ProjectsSearch.tsx

```diff
@@ -0,0 +1,132 @@
+    const [currentProject, setCurrentProject] = useState<Project | undefined>(undefined);
+    const [currentProjectOwner, setCurrentProjectOwner] = useState<string | undefined>("");
+
+    useEffect(() => {
+        const projectId = location.pathname.split('/')[3];
+        if (projectId && searchResult) {
+            let currentProject = searchResult.rows.find(project => project.id === projectId);
+            if (currentProject) {
+                setCurrentProject(currentProject);
+            } else {
+                getGitpodService().server.adminGetProjectById(projectId)
+                    .then(project => setCurrentProject(project))
+                    .catch(e => console.error(e));
+            }
+        } else {
+            setCurrentProject(undefined);
+        }
+    }, [location]);
+
+    useEffect(() => {
+        (async () => {
+            if (currentProject) {
+                if (currentProject.userId) {
+                    const owner = await getGitpodService().server.adminGetUser(currentProject.userId);
+                    if (owner) { setCurrentProjectOwner(owner?.name) }
+                }
+                if (currentProject.teamId) {
+                    const owner = await getGitpodService().server.adminGetTeamById(currentProject.teamId);
+                    if (owner) { setCurrentProjectOwner(owner?.name) }
+                }
+            }
+        })();
+    }, [currentProject])
+
+    if (!user || !user?.rolesOrPermissions?.includes('admin')) {
+        return <Redirect to="/" />
+    }
+
+    if (currentProject) {
+        return <ProjectDetail project={currentProject} owner={currentProjectOwner} />;
+    }
+
+    const search = async () => {
+        setSearching(true);
+        try {
+            const result = await getGitpodService().server.adminGetProjectsBySearchTerm({
+                searchTerm,
+                limit: 50,
+                orderBy: 'creationTime',
+                offset: 0,
+                orderDir: "desc"
+            })
+            setSearchResult(result);
+        } finally {
+            setSearching(false);
+        }
+    }
+
+    return <>
+        <div className="pt-8 flex">
+            <div className="flex justify-between w-full">
+                <div className="flex">
+                    <div className="py-4">
+                        <svg className={searching ? 'animate-spin' : ''} width="16" height="16" fill="none" xmlns="http://www.w3.org/2000/svg">
+                            <path fillRule="evenodd" clipRule="evenodd" d="M6 2a4 4 0 100 8 4 4 0 000-8zM0 6a6 6 0 1110.89 3.477l4.817 4.816a1 1 0 01-1.414 1.414l-4.816-4.816A6 6 0 010 6z" fill="#A8A29E" />
+                        </svg>
+                    </div>
+                    <input type="search" placeholder="Search Projects" onKeyDown={(k) => k.key === 'Enter' && search()} onChange={(v) => { setSearchTerm(v.target.value) }} />
```

Comment:
**issue(non-blocking)**: The issue about state handling that @jldec mentioned in https://github.com/gitpod-io/gitpod/pull/7882#issuecomment-1025706941 is valid but also visible when searching users or workspaces. If this is an online-fix let's include it here but probably not worth blocking this PR as it seems to be out of the scope of the changes here. 🏀

Reply:
ok - agree.